app/src/main/java/comp3350/mealbuddy/objects/MicroFactory.java

```diff
@@ -0,0 +1,25 @@
+package comp3350.mealbuddy.objects;
+
+import java.io.InterruptedIOException;
+import java.util.EnumMap;
+import java.util.Map;
+
+public class MicroFactory {
+
+    public enum Micros{
+        Iron, Zinc, VitaminA, VitaminB12, VitaminC, Choline
+    }
+
+    private MicroFactory(){}    //dont want to instantiate
+
+    public static Map<Micros, Integer> makeMicros(){
+        Map<Micros, Integer> microMap = new EnumMap<>(Micros.class);
+        Micros[] microArr = Micros.values();
+        for (Micros micro:microArr) {
+            microMap.put(micro, 0);
+        }
+        return microMap;
+    }
```

Comment:
I think we need to throw out the enummap and discuss how we want to do this.

an enummap is meant so that we can point enum states to any object, in our case, we just want it to point to an integer.. which is an enum... We need to find a better way to do this, because an enumMap aint it.